numpyro/distributions/random.py

```diff
@@ -12,27 +12,33 @@
 PRNGKey = random.PRNGKey
 
 
-def bernoulli(key, p=0.5, shape=(), dtype=np.float64):
+def bernoulli(key, p=0.5, shape=()):
+    dtype = np.zeros((0,)).dtype
     return random.bernoulli(key, lax.convert_element_type(p, dtype), shape)
```

Comment:
Can we pass in `dtype=float64` here always? If `float64` is enabled that should initialize the sampler with the correct dtype, otherwise it should fall back to `float32`. Alternatively, we can do this just once and use the default type rather than creating a fresh numpy array each time a sampler is called to get the dtype (maybe we don't even need to do this if we can read the jax config instead).

```python
_DEFAULT_DTYPE = None  # depending on when the config is updated we can probably just assign the 
                       # dtype here or check from jax config.

def _get_default_dtype():
    global _DEFAULT_DTYPE
    if not _DEFAULT_DTYPE: 
        _DEFAULT_DTYPE = np.zeros((0,)).dtype
    return _DEFAULT_DTYPE
```

Reply:
It also seems to me that this should be fixed in JAX to ensure that samplers behave consistently w.r.t. other array constructors in that if `float64` mode is enabled they do the same rather than always using `float32` as default.

Reply:
Yes, we can except for normal where finfo(dtype) is used in the implementation. Implementations which depends in `normal` also get this issue.

Reply:
I can follow your second suggestion instead. Thanks!

Reply:
Created a jax issue as well: https://github.com/google/jax/issues/756.

Reply:
Thanks @neerajprad !